File: system-test/dotnet/Driver.Port/Shop/Dtos/BrowseCouponsResponse.cs
namespace Driver.Port.Shop.Dtos;

public class BrowseCouponsResponse
{
    public required List<BrowseCouponItemResponse> Coupons { get; set; }
}

public class BrowseCouponItemResponse
{
    public required string Code { get; set; }
    public required decimal DiscountRate { get; set; }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/port/shop/dtos/BrowseCouponsResponse.java
package com.optivem.shop.dsl.driver.port.shop.dtos;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BrowseCouponsResponse {
    private List<BrowseCouponsItemResponse> coupons;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class BrowseCouponsItemResponse {
        private String code;
        private BigDecimal discountRate;
    }
}
